### ISSUE TITLE
ExactHamiltonianPass inconvenience 

### ISSUE THREAD

**Issue by lfarv** [atcollab/at#176]
Looking at the recently added `ExactHamiltonianPass`, I found its use less convenient than expected:
- it needs  a modification of the elements by requiring an additional field "`Type`",
- it is even worse for drifts, where `NumIntSteps`, `MaxOrder`, `PolynomA`, `PolynomB` are required  though they make no sense there.

I see 2 solutions to make it easier (which can implemented together):

1. A backward compatible modification of `ExactHamiltonianPass`: `Type` can be deduced from the element Class, in the Matlab/Python interface function of `ExactHamiltonianpass.cc`, so that it does not have to be present in the element object. At the same time, only the necessary fields can be extracted, depending on the `Type` value.
2. since this `Type` variable is used as a switch to trigger different computations, one could provide 3 new integrators (`ExactDriftPass`, `ExactBendPass`, `ExactMPolePass` for instance) which can call directly the corresponding computation (no code duplication needed). This is more the AT style…

What do you think about that?

**Comment by willrogers** [comment]
Yes, this makes sense. I'll have a think about it when I next look at AT.
